### DISCUSSION TITLE
Upstream patch(es) status

### DISCUSSION THREAD

**Discussion by shipmints** [d12frosted/homebrew-emacs-plus#763] [General]
@jimeh has adopted the patches from this repo in his very nice (and now nix-based) builds over at
https://github.com/jimeh/emacs-builds and https://github.com/jimeh/build-emacs-for-macos

Per https://github.com/jimeh/emacs-builds/issues/36 we'd like to understand the status of the very good patch set in upstream Emacs. Does anyone know the status of each patch, if it was ever submitted, if it was rejected (and why), if it was accepted but still in the queue (and who is assigned). Perhaps we can help with upstream adoption. We've seen several very good mac/NS patches collaborated on and accepted in recent times.

TIA,

-Stephane

**Comment by d12frosted** [comment]
Hey @shipmints 

Well, accepted patches are not longer part of this repository / latest builds. As of Emacs 31 we have only 3 patches:

- [fix-window-role.patch](https://github.com/d12frosted/homebrew-emacs-plus/blob/master/patches/emacs-31/fix-window-role.patch) - this is mostly needed for yabai tiling window manager;
- [round-undecorated-frame.patch](https://github.com/d12frosted/homebrew-emacs-plus/blob/master/patches/emacs-31/round-undecorated-frame.patch) - allows to have macos-style round corners without any decoration (by default you get 90° corners)
- [system-appearance.patch](https://github.com/d12frosted/homebrew-emacs-plus/blob/master/patches/emacs-31/system-appearance.patch) authored by Nicolas G. Querol. Allows to inherit system appearance.

Personally, I never proposed these patches to the upstream, mostly because upstream doesn't care much about mac-only features and lack of time/motivation. I am not following the devel mailing list, so can't say if any of these patches were ever discussed. I don't mind if someone proposes them to the upstream. AFAIR I have signed documents with FSF (so first two patches can be merged without any issues), but not sure about Nicolas G. Querol.

**Reply by shipmints** [reply]
My recent experience sitting on (and occasionally contributing) to emacs-devel and bug-gnu-emacs suggests that people are indeed eager to improve macOS as a primary Emacs platform. People seem very approachable and since these patches are in the wild and you can probably estimate their uptake / number of users, that might help the maintainers understand the positive impact of these patches. If the patches are truly esoteric, I can imagine some discussion; e.g., yabai use case, BUT if the change is actually beneficial in general to all macOS users, which this one might very well be, it seems like it's worth the effort.

**Reply by d12frosted** [reply]
Oh, https://github.com/d12frosted/homebrew-emacs-plus/pull/776 reminded me about this thread :) There are policies that disallow improving Emacs for macos unless the same improvement can be brought to free systems. This can be interpreted in many ways though hehe. Without sharing my stance/opinion on this policy (it doesn't really matter), it's one of the reasons why I don't have any motivation to contribute, especially considering lack of time. But you see @xenodium is a die hard 😸 

So anyways, just a reminder - I am absolutely open to someone taking the chore of communication regarding these patches.

**Reply by shipmints** [reply]
Been following along. Alvaro @xenodium is a champ. The purist dogma of some people can truly alienate others, for sure. That said, I think your patches fall into the category of achieving parity with "free" shite like wayland, etc. Wayland still doesn't allow apps to position their own windows FFS for "security" as if adults can't think for themselves--a "free" nanny state--and maybe the dogmatics will suggest Windows and Mac users be forced to disable frame positioning to enforce parity with the nannies.
